src/main/java/Duke.java

```diff
@@ -1,10 +1,195 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class Duke {
-    public static void main(String[] args) {
+
+    public static void printLine(){
+        System.out.println("------------------------------------------------------" +
+                "--------------------------");
+    }
+
+    public static void greet(){
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printLine();
+        System.out.println("Hello! I'm Duke\n" + "What can I do for you?");
+        printLine();
+    }
+
+    public static void bye(){
+        printLine();
+        System.out.println("Bye. Hope to see you again soon!");
+        printLine();
+    }
+
+    private static int initialisation(ArrayList<Task> t, int index){
+
+        try {
+            index = readFile("data/data.txt", t);
+        }
+        catch (FileNotFoundException e){
+            System.out.println("File not found");
+        }
+        return index;
+    }
+
+    private static int readFile(String filePath, ArrayList<Task> t)
+            throws FileNotFoundException {
+
+        File f = new File(filePath);
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        System.out.println("yes");
+        int index = 0;
+
+        while (s.hasNext()) {
+
+            String temp = s.nextLine();
+            System.out.println(temp);
+            Task newTask = null;
+
+            if (temp.startsWith("[T]")){
+                newTask = new Todo(temp.substring(7));
+                t.add(newTask);
+            }
+
+            else if (temp.startsWith("[E]")){
+                newTask = new Event(temp.substring(7, temp.indexOf(" (")),
+                        temp.substring(temp.indexOf("(at:") + 5, temp.indexOf(")")));
+                t.add(newTask);
+            }
+
+            else if (temp.startsWith("[D]")){
+                newTask = new Deadline(temp.substring(7, temp.indexOf(" (")),
+                        temp.substring(temp.indexOf("(by:") + 5, temp.indexOf(")")));
+                t.add(newTask);
+            }
+
+            index++;
+        }
+        return index;
+    }
+
+
+    public static int addTask(ArrayList<Task> t, String message, int index) throws DukeException {
+
+        // list
+        if (message.equals("list")){
```

Comment:
need a space between `)` and `{` here.

Tip: find out how to correct these spacing issues on the IDE automatically!